<launch>

  <!--node pkg="csv_to_map" type="add_points_in_localmap.py" name="add_points_in_localmap" output="screen" /-->

  <node pkg="teb_local_planner_tutorials" type="highpass" name="highpass" />

  <node pkg="teb_local_planner_tutorials" type="collision_check" name="collision_check">
    <!--jaundae-->
    <!--param name="goal_x" value="-677.5085" />
    <param name="goal_y" value="4643.3402" /-->

    <!--jaundae parking lot-->
    <!--param name="goal_x" value="-376.689" />
    <param name="goal_y" value="4660.43" /-->

    <!--k_city-->
    <param name="goal_x" value="75.5867" />
    <param name="goal_y" value="211.17" />
    <param name="goal_theta" value="3.054" />

    <!--kaist parking lot-->
    <!--param name="goal_x" value="320.8867" />
    <param name="goal_y" value="-845.8" /-->

    <param name="collision_continuation_threshold" value="10" />
    <!--param name="collision_distance_from_path" value="1.0" /-->
    <!--param name="collision_distance_from_path" value="0.6" /-->
    <param name="collision_distance_from_path" value="1.7" />
    <param name="collision_distance_from_car" value="15.0" />

    <param name="accident_mission_num" value="6" />
  </node>

  <node pkg="teb_local_planner_tutorials" type="publish_goal.py" name="publish_goal" />

  <param name="/use_sim_time" value="false"/>

  <node pkg="move_base" type="move_base" respawn="false" name="move_base">
    <rosparam file="$(find teb_local_planner_tutorials)/cfg/i30_drive/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find teb_local_planner_tutorials)/cfg/i30_drive/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find teb_local_planner_tutorials)/cfg/i30_drive/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find teb_local_planner_tutorials)/cfg/i30_drive/global_costmap_params.yaml" command="load" />
    <rosparam file="$(find teb_local_planner_tutorials)/cfg/i30_drive/teb_local_planner_params.yaml" command="load" />
    <param name="TebLocalPlannerROS/include_costmap_obstacles" value="True" />
    <param name="TebLocalPlannerROS/include_dynamic_obstacles" value="True" />

    <param name="base_global_planner" value="navfn/NavfnROS" />
    <param name="planner_frequency" value="30.0" />
    <param name="planner_patience" value="10.0" />

    <param name="base_local_planner" value="teb_local_planner/TebLocalPlannerROS" />
    <param name="controller_frequency" value="30.0" />
    <param name="controller_patience" value="10.0" />
    <!--remap from="map" to="/map" /-->
  </node>

  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find teb_local_planner_tutorials)/cfg/rviz_navigation_cc.rviz">
    <remap from="move_base_simple/goal" to="/rviz_goal"/>
  </node>

</launch>
